servers/physics_2d/body_2d_sw.cpp

```diff
@@ -384,11 +384,20 @@ void Body2DSW::set_space(Space2DSW *p_space) {
 
 void Body2DSW::_compute_area_gravity_and_dampenings(const Area2DSW *p_area) {
 	if (p_area->is_gravity_point()) {
-		if (p_area->get_gravity_distance_scale() > 0) {
-			Vector2 v = p_area->get_transform().xform(p_area->get_gravity_vector()) - get_transform().get_origin();
-			gravity += v.normalized() * (p_area->get_gravity() / Math::pow(v.length() * p_area->get_gravity_distance_scale() + 1, 2));
+		const real_t gravity_distance_scale = p_area->get_gravity_distance_scale();
+		Vector2 v = p_area->get_transform().xform(p_area->get_gravity_vector()) - get_transform().get_origin();
+		if (gravity_distance_scale > 0) {
+			real_t v_length = v.length();
+			if (v_length > 0) {
+				Vector2 new_gravity = gravity + (v.normalized() * (p_area->get_gravity() / Math::pow(v_length * gravity_distance_scale, 2)));
+				if (isinf(new_gravity.x) || isinf(new_gravity.y)) {
```

Comment:
This check for infinity values is not needed anymore since you're already checking `v_length` and `gravity_distance_scale` are both > 0, so you can just remove the condition and replace with `gravity += ...`.